File: backend/app/models/team_column.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, JSON, Boolean, text
from sqlalchemy.orm import relationship

from app.database import Base


class TeamColumn(Base):
    """팀 칼럼 - 팀원들이 자유롭게 작성하는 글"""
    __tablename__ = "team_columns"

    id = Column(Integer, primary_key=True, index=True)
    author_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    title = Column(String(200), nullable=False)
    content = Column(Text, nullable=True)  # legacy markdown (하위 호환)
    blocks = Column(JSON, nullable=True)  # Editor.js 블록 형식
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # 검증 관련 필드 (팀장/관리자가 파란 체크마크)
    is_verified = Column(Boolean, default=False)
    verified_by = Column(Integer, ForeignKey("users.id"), nullable=True)
    verified_at = Column(DateTime, nullable=True)

    # 방패 적립 여부 (검증 시 미출석이 없어서 방패가 적립된 경우 True)
    shield_granted = Column(Boolean, default=False, nullable=False, server_default=text('false'))

    # Relationships
    author = relationship("User", back_populates="columns", foreign_keys=[author_id])
    verifier = relationship("User", foreign_keys=[verified_by])

File: spring-backend/backend/src/main/java/com/fundmessenger/column/entity/TeamColumn.java
package com.fundmessenger.column.entity;

import com.fundmessenger.user.entity.User;
import io.hypersistence.utils.hibernate.type.json.JsonType;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.Type;

import java.time.OffsetDateTime;

@Entity
@Table(name = "team_columns")
@Getter
@Setter
@NoArgsConstructor
public class TeamColumn {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "author_id", nullable = false, insertable = false, updatable = false)
    private Long authorId;

    @Column(name = "title", length = 200, nullable = false)
    private String title;

    @Column(name = "content", columnDefinition = "text")
    private String content;

    @Type(JsonType.class)
    @Column(name = "blocks", columnDefinition = "jsonb")
    private Object blocks;

    @Column(name = "created_at", columnDefinition = "timestamptz")
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", columnDefinition = "timestamptz")
    private OffsetDateTime updatedAt;

    @Column(name = "is_verified", columnDefinition = "boolean default false")
    private Boolean isVerified = false;

    @Column(name = "verified_by", insertable = false, updatable = false)
    private Long verifiedById;

    @Column(name = "verified_at", columnDefinition = "timestamptz")
    private OffsetDateTime verifiedAt;

    @Column(name = "shield_granted", nullable = false, columnDefinition = "boolean default false")
    private Boolean shieldGranted = false;

    // --- ManyToOne Relationships ---

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private User author;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "verified_by")
    private User verifier;
}
